### ISSUE TITLE
Freshclam stopped updating after last update

### ISSUE THREAD

**Issue by lugdunum-1964** [docker-mailserver/docker-mailserver#2942]
### Miscellaneous first checks

- [X] I checked that all ports are open and not blocked by my ISP / hosting provider.
- [X] I know that SSL errors are likely the result of a wrong setup on the user side and not caused by DMS itself. I'm confident my setup is correct.

### Affected Component(s)

freshclam updates of antivirus database

### What happened and when does this occur?

```Markdown
I received a mail from the clamav system that freshclam had failed
```

### What did you expect to happen?

```Markdown
freshclam retrieving a database update
```

### How do we replicate the issue?

```Markdown
1. I updated the mailserver:
   docker-compose down; docker-compose pull; docker-compose up --build -d
2. 
3.
...
```

### DMS version

v11.3

### What operating system is DMS running on?

Linux

### Which operating system version?

Debian GNU/Linux 11

### What instruction set architecture is DMS running on?

ARM64 / ARM v8 (and newer)

### What container orchestration tool are you using?

Docker

### docker-compose.yml

```yml
version: '3.3'

services:
  mailserver:
    image: docker.io/mailserver/docker-mailserver:latest
    hostname: ${HOSTNAME}
    domainname: ${DOMAINNAME}
    container_name: ${CONTAINER_NAME}
    env_file: mailserver.env

    # To avoid conflicts with yaml base-60 float, DO NOT remove the quotation marks.
    # More information about the mailserver ports:
    # https://docker-mailserver.github.io/docker-mailserver/edge/config/security/understanding-the-ports/
    ports:
      - "25:25"     # SMTP  (explicit TLS => STARTTLS)
      # "143:143"   # IMAP4 (explicit TLS => STARTTLS)  <-- disabled
      - "465:465"   # ESMTP (implicit TLS)
      - "587:587"   # ESMTP (explicit TLS => STARTTLS)
      - "993:993"   # IMAP4 (implicit TLS)
      - "4190:4190" # ManageSieve
    volumes:
      - "./volumes/maildata:/var/mail"
      - "./volumes/mailstate:/var/mail-state"
      - "./volumes/maillogs:/var/log/mail"
      - "/etc/localtime:/etc/localtime:ro"
      - "./config/:/tmp/docker-mailserver/${SELINUX_LABEL}"              # <-- local (re)configuration
      - "/mnt/proxy/certs/:/etc/letsencrypt"                             # <-- SSL certificates
      - "/mnt/proxy/config/dhparams.pem/:/etc/postfix/dhparams.pem:ro"   # <-- TLS crypto file
      - "/mnt/proxy/config/dhparams.pem/:/etc/dovecot/dh.pem:ro"         # <-- same TLS crypto file

    restart: unless-stopped
    stop_grace_period: 1m
    cap_add: [ "NET_ADMIN", "SYS_PTRACE" ]
```

### Relevant log output

```bash
Sun Dec 18 12:00:01 2022 -> --------------------------------------
Sun Dec 18 12:00:01 2022 -> ERROR: Can't create freshclam.dat in /var/lib/clamav
Sun Dec 18 12:00:01 2022 -> Hint: The database directory must be writable for UID 106 or GID 108
Sun Dec 18 12:00:01 2022 -> ERROR: Failed to save freshclam.dat!
Sun Dec 18 12:00:01 2022 -> WARNING: Failed to create a new freshclam.dat!

ad infinitum, ad nauseam
```

### Other relevant information

I repaired it by logging in to the mailserver container and 
`chgrp 108 /var/lib/clamav; chmod g+w /var/lib/clamav.`

But that won't survive a stop and start. I could add a volume to make the changes persistent. But
it probably should be handled in the Dockerfile.

### What level of experience do you have with Docker and mail servers?

- [ ] I am inexperienced with docker
- [X] I am rather experienced with docker
- [ ] I am inexperienced with mail servers
- [X] I am rather experienced with mail servers
- [ ] I am uncomfortable with the CLI
- [X] I am rather comfortable with the CLI

### Code of conduct

- [X] I have read this project's [Code of Conduct](https://github.com/docker-mailserver/docker-mailserver/blob/master/CODE_OF_CONDUCT.md) and I agree
- [X] I have read the [README](https://github.com/docker-mailserver/docker-mailserver/blob/master/README.md) and the [documentation](https://docker-mailserver.github.io/docker-mailserver/edge/) and I searched the [issue tracker](https://github.com/docker-mailserver/docker-mailserver/issues?q=is%3Aissue) but could not find a solution

**Comment by polarathene** [comment]
> I repaired it by logging in to the mailserver container and `chgrp 108 /var/lib/clamav; chmod g+w /var/lib/clamav.`
> 
> But that won't survive a stop and start. I could add a volume to make the changes persistent. But it probably should be handled in the Dockerfile.

In the meantime, you can use [`user-patches.sh`](https://docker-mailserver.github.io/docker-mailserver/edge/config/advanced/override-defaults/user-patches/).

**Comment by lugdunum-1964** [comment]
Good one, didn't think of that :-/ 
cheers!

**Comment by polarathene** [comment]
I had a quick look at this, and for me `/var/lib/clamav` is showing as owned by `clamav(106):clamav(108)`, but the file contents `bytecode.cvd`, `daily.cvd` and `main.cvd` are all `_apt(100):ssl-cert(101)`, which I suspect are just the UID/GID of the files copied over from the official image.

We started [copying over the ClamAV files this way in our builds instead of running `freshclam`](https://github.com/docker-mailserver/docker-mailserver/commit/32c508aa2a4d8899a3c95235650956596d97049f)) during an image build [since the `v11.2` release](https://github.com/docker-mailserver/docker-mailserver/releases/tag/v11.2.0).

---

However when you mount a volume to `/var/mail-state` and our default `ONE_DIR=1`, some extra code is run that maps `/var/lib/clamav` as a root owned symlink to `/var/mail-state/lib-clamav` (owned by clamav user and group):

https://github.com/docker-mailserver/docker-mailserver/blob/8b36e903a21623db7c667d11ca32486328d44fef/target/scripts/startup/misc-stack.sh#L64

This updated the user ownership to `clamav` but not the group.

---

I was able to run `freshclam` without issue regardless of the volume mount though. I could not find the same log errors you're experiencing (_where exactly are you getting those logs from? I checked various logs on my end in `/var/log/supervisor` and `/var/log/mail`_). I run `freshclam` and get output in `/var/log/mail/freshclam.log`, everything looks fine.

There is a [cron job that perform a `freshclam` update every 6 hours roughly](https://github.com/docker-mailserver/docker-mailserver/blob/8b36e903a21623db7c667d11ca32486328d44fef/Dockerfile#L35). I modified the script to run every minute instead, and did `pkill cron`, the cron job was then performed and log looked fine again.

I assume this issue is only happening from the upgrade from a previous version with `/var/mail-state` mounted? Are you able to reproduce it without that volume?

If I understood the issue correctly, you could also modify the permissions of `/var/mail-state/lib-clamav` files, and that shouldn't reset? It should [detect mounted content and just symlink to it](https://github.com/docker-mailserver/docker-mailserver/blob/8b36e903a21623db7c667d11ca32486328d44fef/target/scripts/startup/misc-stack.sh#L48), otherwise it [moves the content in the docker image and creates the symlink](https://github.com/docker-mailserver/docker-mailserver/blob/8b36e903a21623db7c667d11ca32486328d44fef/target/scripts/startup/misc-stack.sh#L53) which is what should happen the first time you mount a volume.

**Comment by polarathene** [comment]
I have made a more minimal Dockerfile with debian and clamav, and noticed that the UID + GID is different. We don't set an explicit numeric value AFAIK, and it seems that this changes based on the order of the package being installed.

In my minimal version the UID is 101 and GID is 102. On storage side, it's the numeric ID, and your host OS and the Docker container will just match those values to what is found in `/etc/passwd` and `/etc/group` AFAIK, so anything that looks for a `clamav` user/group and defers that user/group ID may break when our packages have changed between releases and the values stored in your volume differ from what the containers numeric mapping for `clamav` user and group IDs are :sweat_smile: 

In the minimal Dockerfile I used (_same base image in the projects Dockerfile, but stripping away anything not ClamAV related, only installs two ClamAV packages, then does the linked COPY from official ClamAV image_), the user 100 remains `_apt` but group 101 is now `crontab` instead of `ssl-cert`, while `clamav` has UID 101 and GID 102.
